include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -256,14 +298,49 @@ struct tuple<T1, T...>
 
     using tuple_type = ::std::tuple<T1, T...>;
 
+    template <::std::size_t I>
+    constexpr auto
+    get() & -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const& -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() && -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const&& -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
     tuple() = default;
     tuple(const tuple& other) = default;
+    tuple(tuple&& other) = default;
     template <typename U1, typename... U>
     tuple(const tuple<U1, U...>& other) : holder(other.holder.value), next(other.next)
     {
     }
 
-    tuple(const T1& _value, const T&... _next) : holder(_value), next(_next...) {}
+    template <typename _U1, typename... _U,
+              typename = typename ::std::enable_if<
+                  (sizeof...(_U) == sizeof...(T) && ::std::is_constructible<T1, _U1&&>::value &&
+                   oneapi::dpl::__internal::__conjunction<::std::is_constructible<T, _U&&>...>::value)>::type>
+    tuple(_U1&& _value, _U&&... _next) : holder(::std::forward<_U1>(_value)), next(::std::forward<_U>(_next)...)
```

Comment:
For which function we need this ctor?

Reply:
The problem is previously we worked with `const` lvalue only. If we want to support rvalue category we need also to preserve that, which was impossible with previous constructor.

Reply:
Kind of generalization, yes?

Reply:
I am not sure what you mean by generalization but basically yes, we support all the value categories